app/controllers/users/auth_app_controller.rb

```diff
@@ -0,0 +1,66 @@
+module Users
+  class AuthAppController < ApplicationController
+    include ReauthenticationRequiredConcern
+
+    before_action :confirm_two_factor_authenticated
+    before_action :confirm_recently_authenticated_2fa
+    before_action :set_form
+    before_action :validate_configuration_exists
+
+    def edit; end
+
+    def update
+      binding.pry
+      result = form.submit(name: params.dig(:form, :name))
+
+      analytics.auth_app_update_name_submitted(**result.to_h)
+
+      if result.success?
+        flash[:success] = t('two_factor_authentication.auth_app.renamed')
+        redirect_to account_path
+      else
+        flash.now[:error] = result.first_error_message
+        render :edit
+      end
+    end
+
+    def destroy
+      result = form.submit
+
+      analytics.auth_app_delete_submitted(**result.to_h)
+
+      if result.success?
+        flash[:success] = t('two_factor_authentication.webauthn_platform.deleted')
+        create_user_event(:webauthn_key_removed)
+        revoke_remember_device(current_user)
+        event = PushNotification::RecoveryInformationChangedEvent.new(user: current_user)
+        PushNotification::HttpPush.deliver(event)
+        redirect_to account_path
+      else
+        flash[:error] = result.first_error_message
+        redirect_to edit_auth_app_path(id: params[:id])
+      end
+    end
+
+    private
+
+    def form
+      @form ||= form_class.new(user: current_user, configuration_id: params[:id])
+    end
+
+    alias_method :set_form, :form
+
+    def form_class
+      case action_name
+      when 'edit', 'update'
+        TwoFactorAuthentication::AuthAppUpdateForm
+      when 'destroy'
+        TwoFactorAuthentication::AuthAppDeleteForm
+      end
+    end
```

Comment:
I think this abstraction is more difficult to follow and it would be preferable to explicitly instantiate the form in each of the controllers. It would also allow dropping the `set_form` alias and before_action.

Reply:
With `form_class`, we use that method to define the form in each controller and the `validate_configuration_exists` before_action. Wouldn't dropping `set_form` break the validation before_action?

Reply:
Ah, I see what you mean. It would break that, but I think it may be worth splitting it up still and being explicit about that too since this makes it difficult to respond with something other than `render_not_found`. The existing controller redirects, and I think that might be preferable (perhaps with an error message?).

Reply:
After some thought, I will stick to what I am proposing--we are using one view to rename and delete an app method and the form that we need in the view is toggled based on the desired action.